config/site/site.config.js

```diff
@@ -0,0 +1,111 @@
+const i18nSupportedLocales = require('terra-i18n/lib/i18nSupportedLocales');
+const navConfig = require('./navigation.config');
+const startCase = require('lodash.startcase');
+const fs = require('fs');
+const path = require('path');
+
+const npmPackage = JSON.parse(fs.readFileSync(path.resolve(process.cwd(), 'package.json')));
+
+const siteConfig = {
+  /* The navigation configuration.  */
+  navConfig,
+
+  /* The path to the pages configuration. If this is enabled, page config will not be generated. */
+  // pagesConfig,
+
+  // These options are used to find the pages to serve via terra dev site.
+  // If 'pagesConfig' is provided, no pages will be generated.
+  // Output:
+  // {root}/lib/{entryPoint}/**/*{pageType}.{jsx,js,md,}
+  // {root}/packages/*/lib/{entryPoint}/**/*{pageType}.{jsx,js,md,}
+  // source will be used if hot reloading is enabled and not running in prod mode.
+  // source and dist are optional.
+  // *NOTE* The entryPoints also act as the beginning of the directory strucure for menu navigation.
+  // For example: /root/src/<entryPointDir>/folder/item would start navigation at 'folder'.
+  generatePages: {
+    searchPatterns: [
+      {
+        root: process.cwd(),
+        source: 'src',
+        dist: 'lib',
+        entryPoint: 'terra-dev-site',
+      },
+    ],
+  },
+
+  /* Hot reloading section. On by default for dev builds. This will automatically alias the dist dir as the source dir if you use the bundled webpack config. */
+  hotReloading: {
+    enabled: true,
+    source: 'src',
+    dist: 'lib',
+  },
+
+  /* The parsed root level npm package. May need to change this if you have a non-standard package.json path. */
+  npmPackage,
+
+  /* Side effect theme imports */
+  themeImports: [
+  //   './theme';
+  ],
+
+  /* Path to the image to display as page placeholder when a component does not render. */
+  placeholderSrc: path.join(__dirname, '..', '..', 'terra.png'),
+
+  /* The README content to display on the home page. */
+  readMeContent: path.resolve(process.cwd(), 'README.md'),
+
+  appConfig: {
+    /* Path to the logo the site header should display. */
+    logoSrc: path.join(__dirname, '..', '..', 'terra.png'),
+
+    /* The title the site header should display. */
+    title: startCase(npmPackage.name), // maps to appTitle
+
+    /* Path to the favicon for the site. */
+    favicon: path.join(__dirname, '..', '..', 'terra-favicon', '32px', 'favicon.ico'),
```

Comment:
Well you just stole one of the first things I was going to have Nathan do 😄 Glad we are finally adding this.

Reply:
Make nathan add search.